agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,188 @@
+    sleep 1
+    read -p "Another tool? (y/n)" anothertool
+done
+printf -- "\nDone registering tools.\n\n"
+ls -lR ${pbench_run}/tools-v1-default
+
+printf -- "\n\n"
+wait_keypress 120
+
+
+# Start Tool Meister containers on remote host
+
+printf -- "\n\nNow please start the Tool Meister container on the specified host...\n\n\t$ podman run --name pbench-agent-tool-meister \\ \n\t\t--network host --ulimit nofile=65536:65536 --rm -d \\ \n\t\t-e REDIS_HOST=${REDIS_HOST} \\ \n\t\t-e REDIS_PORT=${REDIS_PORT} \\ \n\t\t-e PARAM_KEY=tm-default-\$(hostname -f) \\ \n\t\t-e _PBENCH_TOOL_MEISTER_LOG_LEVEL=debug \\ \n\t\tquay.io/pbench/pbench-agent-tool-meister-${DISTRO}:${TAG}\n\n"
+
+wait_keypress 120
+
+
+# Start the Tool Data Sink locally, mapping our volume into the Tool Data Sink
+# container.
+
+printf -- "\n\nNow we will automatically start the local Tool Data Sink to pull tool data into our local volume.\n\n"
+
+set -x
+podman run --name pbench-agent-tool-data-sink --network host --volume ${pbench_run}:/var/lib/pbench-agent:Z --ulimit nofile=65536:65536 --rm -d -e REDIS_HOST=${REDIS_HOST} -e REDIS_PORT=${REDIS_PORT} -e PARAM_KEY=tds-default -e _PBENCH_TOOL_DATA_SINK_LOG_LEVEL=debug quay.io/pbench/pbench-agent-tool-data-sink-${DISTRO}:${TAG}
+set +x
+
+printf -- "\n\nOptional: Now let's look at the 'podman logs' output from these containers;\n\t You will notice we did not start a Redis server yet,\n\tthey'll just be waiting for it to show up.\n\n"
+
+wait_keypress 120
+
+
+printf -- "\n\nPlease start the Redis server on ${REDIS_HOST}:${REDIS_PORT}\n\t$ podman run --name demo-tm-redis --network host --rm -d redis\n\n\tNote TDS and TMs notice Redis server,\n\tbut now wait for their 'PARAM_KEY' to show up.\n\n"
+
+wait_keypress 120
+
+export date="2021-1-1T12:00:00"
+
+source /etc/profile.d/pbench-agent.sh
+source /opt/pbench-agent/base
+
+group="default"
+
+export script="demo"
+export config="my-demo-config-000"
+export benchmark_run_dir="${pbench_run}/${script}_${config}_${date_suffix}"
+mkdir ${benchmark_run_dir}
+
+printf -- "\n\nTypically pbench-tool-meister-start is expecting a '\${benchmark_run_dir}' to store data\nwhich is usually created by a pbench-user-benchmark and the like.\nWe mimic the same behavior with:\n\n\tbenchmark_run_dir='${benchmark_run_dir}'\n\n"
+
+wait_keypress 120
+
+
+# Start the Tool Meisters, collecting system information, and start any persistent tools.
+set -x
+_PBENCH_TOOL_MEISTER_START_LOG_LEVEL=debug pbench-tool-meister-start --sysinfo=default ${group} 2>&1 | less -S
+set +x
+
+printf -- "\n\nThe operation of pbench-tool-meister-start creates the keys containing the operational data for the Tool Data Sink and the Tool Meisters, then issued the first 'sysinfo' collection, as requested, and sent the 'init' persistent tools command.\n\nAt this point any registered persistent tools are up and running. Next is the handling of transient tool start/stop.\n\n"
```

Comment:
The tenses here could be cleaned up. E.g., "The pbench-tool-meister-start command first created the Redis keys to communicate with the Tool Data Sink and Tool Meisters, then issued ..."